integrations/google-analytics/lib/index.js

```diff
@@ -229,12 +230,18 @@ GA.prototype.page = function(page) {
   };
 
   // Reset custom dimension which are previously set.
-  var customDimension = {};
-  var dimensionsKeys = Object.keys(opts.dimensions);
-  for (var i = 0; i < dimensionsKeys.length; i++) {
-    customDimension[dimensionsKeys[i]] = '';
+  // Uses the configured dimensions as:
+  // opts.dimensions: { "fruit": "dimension1" }
+  // opts.resetCustomDimensions: [ "fruit" ]
+  // --> resetCustomDimensions: { "dimension1": null }
+  var resetCustomDimensions = {};
+  for (var i = 0; i < opts.resetCustomDimensionsOnPage.length; i++) {
+    var property = opts.resetCustomDimensionsOnPage[i];
+    if (opts.dimensions[property]) {
+      resetCustomDimensions[opts.dimensions[property]] = null;
+    }
   }
-  window.ga(self._trackerName + 'set', customDimension);
+  window.ga(self._trackerName + 'set', resetCustomDimensions);
```

Comment:
I don't know enough about this integration but looks like you're no longer setting `customDimension`. Is that intentional?

Reply:
Yep, the change is to unset (ie set to `null`) the dimensions specified on the new setting. We do that because by default, Google Analytics keeps the dimensions in memory, and even if you don't send them in a new event, they will get added to the call.

This only happens before a page call, for SPA websites.